test/controllers/users_controller_test.rb

```diff
@@ -10,7 +10,7 @@ class UsersControllerTest < ActionDispatch::IntegrationTest
 
   describe 'create method' do
     it 'should raise if user is not admin' do
-      User::ROLE.reject { |role| role == :admin }.each do |role|
+      User.role.values.reject { |role| role == :admin }.each do |role|
```

Comment:
That's pretty neat, I remember mongoid-enum mentioning how it makes globals accessible everywhere.

And keep the textbook information coming! Normally I'd just ignore it, but there is actually context and I was curious what enumeration was all about. I'm probably more optimistic and think the solution is good enough. But something tells me I am not seeing the side effects that you're anticipating.